src/components/NftGallery/NftGallery.tsx

```diff
@@ -24,36 +24,26 @@ function NftGallery({ collection, mobileLayout }: Props) {
   return (
     <StyledCollectionNfts columns={columns} mobileLayout={mobileLayout}>
       {collection.nfts.map((nft) => (
-        <NftPreview
-          key={nft.id}
-          nft={nft}
-          collectionId={collection.id}
-          columns={columns}
-          mobileLayout={mobileLayout}
-        />
+        <NftPreview key={nft.id} nft={nft} collectionId={collection.id} columns={columns} />
       ))}
     </StyledCollectionNfts>
   );
 }
 
 const StyledCollectionNfts = styled.div<{ columns: number; mobileLayout: DisplayLayout }>`
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: ${({ columns, mobileLayout }) =>
+    mobileLayout === DisplayLayout.LIST ? '1fr' : `repeat(${columns},  minmax(auto, 50%))`};
```

Comment:
what is this black magic

Reply:
its kinda crazy, im learning display:grid is pretty powerful